packages/x-date-pickers/src/ClockPicker/ClockNumber.tsx

```diff
@@ -62,13 +83,7 @@ export function ClockNumber(props: ClockNumberProps) {
 
   return (
     <ClockNumberRoot
-      className={clsx(
-        {
-          [classes.selected]: selected,
-          [classes.disabled]: disabled,
-        },
-        className,
-      )}
+      className={clsx(className, classes.root)}
```

Comment:
L29 covers applying `selected` or `disabled` classes instead as in most other cases.